### ISSUE TITLE
Question about run-trust4

### ISSUE THREAD

**Issue by CoooooolM** [liulab-dfci/TRUST4#338]
Hello and thanks for the great software toll.
When I run the run-trust4 program, I encounter the following error message. How can I solve it?
The following are my commands and error messages:
run-trust4 \
  -f IMGT_vANR.fa \
  --ref IMGT_vANR.fa \
  -1 fastp/IFV-110_1.fastq \
  -2 fastp/IFV-110_2.fastq \
  -o TRUST4-1.1.5 vNAR_IFV

[Mon Dec 23 13:33:04 2024] TRUST4 begins.
[Mon Dec 23 13:33:04 2024] SYSTEM CALL: /home/huang_yongming/miniconda3/bin/fastq-extractor -t 1 -f /home/huang_yongming/software/TRUST4-1.1.5/IMGT_vANR.fa -o /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_toassemble  -1 /home/huang_yongming/vNAR/gene/fastp/IFV-110_1.fastq -2 /home/huang_yongming/vNAR/gene/fastp/IFV-110_2.fastq
[Mon Dec 23 13:33:04 2024] Start to extract candidate reads from read files.
[Mon Dec 23 13:38:44 2024] Finish extracting reads.
[Mon Dec 23 13:38:44 2024] SYSTEM CALL: /home/huang_yongming/miniconda3/bin/trust4  -f /home/huang_yongming/software/TRUST4-1.1.5/IMGT_vANR.fa -o /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV -1 /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_toassemble_1.fq -2 /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_toassemble_2.fq
[Mon Dec 23 13:38:44 2024] SYSTEM CALL: /home/huang_yongming/miniconda3/bin/annotator -f /home/huang_yongming/software/TRUST4-1.1.5/IMGT_vANR.fa -a /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_final.out -t 1 -o /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV  -r /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_assembled_reads.fa > /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_annot.fa
Need to use -a to specify the assembly file.
system /home/huang_yongming/miniconda3/bin/annotator -f /home/huang_yongming/software/TRUST4-1.1.5/IMGT_vANR.fa -a /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_final.out -t 1 -o /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV  -r /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_assembled_reads.fa > /home/huang_yongming/software/TRUST4-1.1.5vNAR_IFV_annot.fa failed: 256 at /home/huang_yongming/miniconda3/bin/run-trust4 line 48.

**Comment by mourisl** [comment]
I think the bioconda TRUST4 version is an old one, otherwise, the first line will output the version number. I think the issue might be that TRUST4 did not identify any candidate reads from the VDJ region and the "xx_toassemble_xxx.fq" files are empty.

**Comment by CoooooolM** [comment]
Is this due to a version problem? So how do I update the version?

> I think the bioconda TRUST4 version is an old one, otherwise, the first line will output the version number. I think the issue might be that TRUST4 did not identify any candidate reads from the VDJ region and the "xx_toassemble_xxx.fq" files are empty.

**Comment by mourisl** [comment]
In the newer version, TRUST4 won't crash when there is no VDJ reads extracted. But you will have an empty output.

**Comment by CoooooolM** [comment]
> In the newer version, TRUST4 won't crash when there is no VDJ reads extracted. But you will have an empty output.

Yes, I got the empty file. Is this due to a problem with my reference genome? What should I do to get the right result?

**Comment by mourisl** [comment]
Can you show me a few lines of your "IMGT_vANR.fa " file? A more common reason for not getting any VDJ reads is because of the low T cell and B cell presence in the sample.

**Comment by CoooooolM** [comment]
> Can you show me a few lines of your "IMGT_vANR.fa " file? A more common reason for not getting any VDJ reads is because of the low T cell and B cell presence in the sample.
Ok, "IMGT_vANR.fa "file is attached. My sequencing sequence is CDR.
[IMGT_vANR.txt](https://github.com/user-attachments/files/18236373/IMGT_vANR.txt)

**Comment by mourisl** [comment]
Oh, I see. Could you please convert the lower-case sequences to upper-case, and give it a try?
